org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -176,6 +184,9 @@ public static Preferences createFrom(Map<String, Object> configuration) {
 		boolean referenceCodelensEnabled = getBooleanValue(configuration, REFERENCES_CODE_LENS_ENABLED_KEY, true);
 		prefs.setReferencesCodelensEnabled(referenceCodelensEnabled);
 
+		boolean signatureHelpEnabled = getBooleanValue(configuration, SIGNATURE_HELP_ENABLED_KEY, true);
+		prefs.setSignatureHelpEnabled(signatureHelpEnabled);
```

Comment:
signaturehelp capability needs to be dynamically (un)registered, similar to formatting